### PR TITLE
[py2py3] Migration at level scr/python/A/B/C - slice 5

### LINKED ISSUE #10116: [py2py3] Migration at level scr/python/A/B/C - slice 5
#### Intro

Our purpose is to continue with the gradual migration. 
This is the 5-th slice in which the migration has been divided into. 
We have split migration units at level `src/python/A/B/C`.

A detailed plan of the migration at this depth is available 
at the dmwm mkdocs [detailed plan page](https://cms-dmwm-test.docs.cern.ch/py2py3/plan_detailed/), 
with summary information on top.

#### Impact of the new feature

We start migrating the 5-th bulk of migration units 
at src/python/A/B/C level.

```plaintext

                                    name  nfiles  gradual  future_pm
           WMCore_Services_UserFileCache       1     True          2
                 WMCore_Services_WMAgent       1     True          0
                    WMCore_Services_WMBS       1     True          0
                  WMCore_Storage_Execute       1     True          1
             WMCore_Storage_StageOutImpl       1     True          3
           WMCore_Storage_StageOutImplV2       1     True          3
                    WMCore_WMBS_WMBSBase       1     True          0
          WMCore_WorkQueue_WorkQueueBase       1     True          3
        WMCore_WebTools_DBSRESTFormatter       1     True          0
                    WMCore_REST_Services       1     True          0
 WMComponent_DBS3Buffer_DBSBufferDataset       1     True          0
               WMCore_Agent_HeartbeatAPI       1     True          1
                  WMCore_Services_ReqMgr       1     True          3
                        WMCore_WMBS_File       1     True          9
                     WMCore_WMBS_Fileset       1     True          1
                        WMCore_WMBS_Mask       1     True          6
                    WMCore_WMBS_Workflow       1     True          2
   WMCore_JobStateMachine_ConfigureState       1     True          0
                    WMCore_WMBS_WorkUnit       1     True          0
              WMCore_Database_CouchUtils       1     True          1
        WMCore_JobStateMachine_SummaryDB       1     True         12
                  WMCore_Services_FWJRDB       1     True          6
                        WMCore_WMLogging       1     True          0
               WMCore_WMSpec_Persistency       1     True         11
              WMCore_GroupUser_Interface       1     True          6
                  WMCore_GroupUser_Group       1     True          0
                   WMCore_GroupUser_User       1     True          0
              WMCore_MicroService_WebGui       1     True          0
                    WMCore_Services_CRIC       1     True          2
          WMQuality_Emulators_CRICClient       2     True          4
                 WMCore_Storage_Registry       1     True          5
               WMCore_Storage_StageInMgr       1     True          6
                 WMCore_Storage_TestImpl       1     True          1
                  WMCore_Database_DBCore       1     True          0
    WMComponent_DBS3Buffer_DBSBufferFile       1     True          0
```

#### Is your feature request related to a problem? Please describe.

The end goal is to have WMCore py2 and py3 compatible.

#### Describe the solution you'd like

Run futurize on the first bulk of units 
and then apply the necessary manual changes.

### PR DESCRIPTION
**Pull request by mapellidario**
Fixes #10116 

#### Status

Ready - waiting final green light

#### Related PRs

* Previous migration PR: #10165 (slice4, issue #10115  )
* Next migration PR: #10277 (slice6, issue  #10127 )

#### Is it backward compatible (if not, which system it affects?)

It should be!

#### External dependencies / deployment changes

no changes

#### Description

Run futurize and some manual changes on the first batch of src/python/A/B/C.

I decided to use the ["native string"](https://portingguide.readthedocs.io/en/latest/strings.html#the-native-string) approach and not force to use `from builtins import str` extensively when not needed. We will see how it goes. This has the advantage of reducing migration effort and without messing with strings when it is not really needed.
This may cause troubles only in:
* `src/python/WMCore/WMBS/Fileset.py ` 
  - did not use `from builtins import str`
  - to be tested in py3, to pen an issue to track this
  - [EDIT] This is not so likely yo cause problems, not opening any issue and hoping that nothing breaks
* `src/python/WMCore/GroupUser/Interface.py`: 
  - using of `from builtins import str`
  - despite it may not be necessary and it may break something
* `test/python/WMCore_t/Database_t/DBCore_t.py`: 
  - not using `from builtins import str`
  - but there is a slight chance that it may cause problems in py3

Dict
* `src/python/WMCore/JobStateMachine/SummaryDB.py`: dict updated in a loop, may be dangerous combined with `viewitems`. Tracked in #10446 

* `DBCore_t`: the following lines have not been changed

    ```python
    def testBuildBinds(self):
        ...
        dictbinds     = {'lumi':123, 'test':100}
        ....
        for i in range(len(files)):
            self.assertEqual(binds[i][seqname], files[i])
            self.assertEqual(binds[i][dictbinds.keys()[0]], dictbinds[dictbinds.keys()[0]])
   ```
   Despite the `pylint --py3k` report warnings
   ```plaintext
        test/python/WMCore_t/Database_t/DBCore_t.py
            W1655, line 64 in DBCoreTest.testBuildBinds: dict.keys referenced when not iterating
            W1655, line 64 in DBCoreTest.testBuildBinds: dict.keys referenced when not iterating
    ```
   A possible solution would be to use the `lumi` instead of `.keys()[0]`, but I would prefer to change the code as little as possible and live with some unharmful warnings, unless told otherwise.

[EDIT] We should fix this in #10398

#### jenkins report

* `pylint --py3k`: 2 warnings considered as false positives
* `pylint`: 24 errors, but i do not feel confident changing those, since it would involve changing interfaces
* `python3 compatibility`: complains about "requirements.txt", no problems in this PR

### CODE REVIEW

**Comment by mapellidario** [comment]
test this, please

**Review by amaltaro — CHANGES_REQUESTED** [review]
Dario, code looks good in general. I have made a few comments along the code though to be taken into consideration.

You mention about Fileset.py in the PR description, but I haven't seen any changes made to that module. Did I miss anything?

One more question/request, could you please check those WMBS/* modules against your oracle jenkins setup?

File: src/python/WMCore/WMSpec/Persistency.py [hunk 1]
```diff
@@ -10,20 +10,20 @@
 """
 from __future__ import print_function
 
-from urllib2 import urlopen, Request
+from future import standard_library
+standard_library.install_aliases()
+from urllib.request import urlopen, Request
```
[the remark this answers is not in the document]
Comment (mapellidario) [reply src/python/WMCore/WMSpec/Persistency.py:15]: You are right! this is going to break cPickle support!
It felt a bit spooky to me that we missed the modernization of this file and the answer was in #9762 PR description! shame on me that simply assumed it was an error of Dario-of-the-past and did look much into possible cause.

Thanks for trusting that past-me wouldn't have missed a file more than I do!

File: src/python/WMCore/Services/ReqMgr/ReqMgr.py [hunk 2]
```diff
@@ -64,8 +66,8 @@ def _createQuery(self, queryDict):
             args = "_nostale=true&"
         else:
             args = ""
-        for name, values in queryDict.items():
-            if isinstance(values, basestring) or isinstance(values, int):
+        for name, values in list(queryDict.items()):
```
[the remark this answers is not in the document]
Comment (mapellidario) [reply src/python/WMCore/Services/ReqMgr/ReqMgr.py:69]: Ops, I have no idea how i missed this!

I would stick with `viewitems`, following [this sugestion in future code](https://github.com/PythonCharmers/python-future/blob/4657ad23de79a541ebcc7a06f1b9ad60172ad3c4/src/future/utils/__init__.py#L310). In py3 both `future.utils.viewitems` and `future.utils.iteritms` load `dict.items`, while in py2 they load `viewitems/iteritems` directly.
As for which is the better, stackoverflow insists that they have similar CPU-time performance [here](https://stackoverflow.com/a/23083560) and [here](https://stackoverflow.com/a/10190228). Memory-wise, I have to come up with some tests, but if they were different I assume that they would have stated it clearly [in the docs](https://python-future.org/compatible_idioms.html#dictionaries). I will come up with some tests however!

File: src/python/WMCore/Services/FWJRDB/FWJRDBAPI.py [hunk 3]
```diff
@@ -71,7 +74,7 @@ def _formatCouchData(self, data, key = "id", detail = True, filterCouch = True,
         if detail or returnDict:
             return result
         else:
-            return result.keys()
+            return list(result.keys())
```
[the remark this answers is not in the document]
Comment (mapellidario) [reply src/python/WMCore/Services/FWJRDB/FWJRDBAPI.py:77]: And again, you re correct! they are exactly the same

File: src/python/WMCore/Services/ReqMgr/ReqMgr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (amaltaro) [reply src/python/WMCore/Services/ReqMgr/ReqMgr.py:69]: Don't worry about testing their performance Dario. Let's just stick with what we have been using and what's suggested (viewitems)

**Comment by mapellidario** [comment]
Test this please

**Comment by mapellidario** [comment]
I run the unit tests in WMBS_t in my VM oracle setup. With this PR I get the same failure as in `master ` there is only one change: `First part of what used to be testRandomCrapForGenerators` does not fail anymore!

I paste here the diff of the unit tests log:

```diff
> diff -u --color unittest_master.txt unittest_pr.txt
--- unittest_master.txt 2021-02-11 12:39:21.711882666 +0100
+++ unittest_pr.txt     2021-02-11 12:39:01.740085287 +0100
@@ -1,4 +1,4 @@
-> python setup.py test --buildBotMode=true --reallyDeleteMyDatabaseAfterEveryTest=true --testCertainPath=test/python/WMCore_t/WMBS_t/        
+> python setup.py test --buildBotMode=true --reallyDeleteMyDatabaseAfterEveryTest=true --testCertainPath=test/python/WMCore_t/WMBS_t/
 running test
 Using the tests below: test/python/WMCore_t/WMBS_t/
 #### WE ARE DELETING YOUR DATABASE. 3 SECONDS TO CANCEL ####
@@ -6,7 +6,7 @@
 We are going to trash databases after every test
 ### We are in buildbot mode ###
 ----------------------------------------------------------------------
-Ran 240 tests in 0.020s
+Ran 240 tests in 0.018s
 
 OK
 path lists is ['test/python/WMCore_t/WMBS_t/']
@@ -54,7 +54,7 @@
 #96 _testLimit_ ... ok
 #97 _testLocationSplit_ ... ok
 #98 _testMoreFiles_ ... ok
-#99 First part of what used to be testRandomCrapForGenerators ... FAIL
+#99 First part of what used to be testRandomCrapForGenerators ... ok
 #107 _testAllAcquired_ ... ok
 #108 _testClosed_ ... ok
 #109 _testClosedSomeAcquired_ ... ok
@@ -353,56 +353,7 @@
 root: DEBUG: Tables for WMCore.WMBS created
 root: INFO: Workflow id 1 created for wf001
 root: INFO: Fileset created: TestFileset
-root: INFO: Fileset created: 9cc299fa-9da2-432f-9cd7-93b8e8cfd410
---------------------- >> end captured logging << ---------------------
-
-======================================================================
-FAIL: First part of what used to be testRandomCrapForGenerators
-----------------------------------------------------------------------
-Traceback (most recent call last):
-  File "/data/cms-dmwm/wmcore_unittest/WMCore/test/python/WMCore_t/WMBS_t/JobSplitting_t/FileBased_t.py", line 515, in testRandomCrap1
-    self.assertEqual(len(jobGroups), 6)
-AssertionError: 8 != 6
--------------------- >> begin captured logging << --------------------
-root: DEBUG: Tables for WMCore.WMBS created
-root: INFO: Fileset created: TestFileset1
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 2}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 3}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 4}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 5}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 6}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 7}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 8}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 9}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 10}]
-root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 11}]
-root: INFO: Fileset created: TestFileset2
-root: INFO: Fileset created: TestFileset3
-root: INFO: Workflow id 1 created for wf001
-root: DEBUG: Opening DB resultProxies for JobFactory
-root: DEBUG: Received 1 proxies
-root: DEBUG: About to load files by proxy
-root: DEBUG: Loaded 2 files
-root: DEBUG: About to commit 1 jobGroups
-root: DEBUG: About to commit 2 jobs
-root: DEBUG: About to load files by proxy
-root: DEBUG: Loaded 2 files
-root: DEBUG: About to commit 2 jobGroups
-root: DEBUG: About to commit 1 jobs
-root: DEBUG: About to load files by proxy
-root: DEBUG: Loaded 2 files
-root: DEBUG: About to commit 2 jobGroups
-root: DEBUG: About to commit 1 jobs
-root: DEBUG: About to load files by proxy
-root: DEBUG: Loaded 2 files
-root: DEBUG: About to commit 1 jobGroups
-root: DEBUG: About to commit 2 jobs
-root: DEBUG: About to load files by proxy
-root: DEBUG: Loaded 2 files
-root: DEBUG: About to commit 2 jobGroups
-root: DEBUG: About to commit 1 jobs
-root: DEBUG: About to load files by proxy
-root: DEBUG: Loaded 0 files
+root: INFO: Fileset created: 8fa6762c-1b5b-470e-be14-f3b1815ab722
 --------------------- >> end captured logging << ---------------------
 
 ======================================================================
@@ -423,7 +374,7 @@
 restkit.client.request: INFO: Start request: HEAD http://localhost:6994/lumi_t
 restkit.client.request: DEBUG: Request headers: [[u'HEAD /lumi_t HTTP/1.1\r\n', 'Host: localhost:6994\r\n', 'User-Agent: couchapp/0.7.1\r\n', 'Accept-Encoding: None\r\n', 'Accept: application/json\r\n', 'Authorization: Basic dW5pdHRlc3RhZ2VudDpwYXNzd2Q=\r\n', '\r\n']]
 restkit.client.request: DEBUG: Start response: 200 OK
-restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Date', 'Wed, 10 Feb 2021 11:06:40 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '225'), ('Cache-Control', 'must-revalidate')]]
+restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Date', 'Wed, 10 Feb 2021 11:50:12 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '225'), ('Cache-Control', 'must-revalidate')]]
 restkit.client.request: DEBUG: Return response: http://localhost:6994/lumi_t
 couchapp.localdoc: DEBUG: push updates/newgroup.js
 couchapp.localdoc: DEBUG: push updates/newuser.js
@@ -442,7 +393,7 @@
 restkit.client.request: INFO: Start request: GET http://localhost:6994/lumi_t/_design/GroupUser
 restkit.client.request: DEBUG: Request headers: [[u'GET /lumi_t/_design/GroupUser HTTP/1.1\r\n', 'Host: localhost:6994\r\n', 'User-Agent: couchapp/0.7.1\r\n', 'Accept-Encoding: None\r\n', 'Accept: application/json\r\n', 'Authorization: Basic dW5pdHRlc3RhZ2VudDpwYXNzd2Q=\r\n', '\r\n']]
 restkit.client.request: DEBUG: Start response: 404 Object Not Found
-restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Date', 'Wed, 10 Feb 2021 11:06:40 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '41'), ('Cache-Control', 'must-revalidate')]]
+restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Date', 'Wed, 10 Feb 2021 11:50:12 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '41'), ('Cache-Control', 'must-revalidate')]]
 restkit.client.request: DEBUG: Return response: http://localhost:6994/lumi_t/_design/GroupUser
 couchapp.localdoc: DEBUG: attach index.html 
 couchapp.localdoc: DEBUG: attach style/main.css 
@@ -459,14 +410,14 @@
 restkit.client.request: INFO: Start request: PUT http://localhost:6994/lumi_t/_design/GroupUser
 restkit.client.request: DEBUG: Request headers: [[u'PUT /lumi_t/_design/GroupUser HTTP/1.1\r\n', 'Host: localhost:6994\r\n', 'User-Agent: couchapp/0.7.1\r\n', 'Accept-Encoding: None\r\n', 'Accept: application/json\r\n', 'Content-Length: 32028\r\n', 'Content-Type: application/json\r\n', 'Authorization: Basic dW5pdHRlc3RhZ2VudDpwYXNzd2Q=\r\n', '\r\n']]
 restkit.client.request: DEBUG: Start response: 201 Created
-restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Location', 'http://localhost:6994/lumi_t/_design/GroupUser'), ('ETag', '"1-c2e1905c915c668f4395b69025da5746"'), ('Date', 'Wed, 10 Feb 2021 11:06:40 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '80'), ('Cache-Control', 'must-revalidate')]]
+restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Location', 'http://localhost:6994/lumi_t/_design/GroupUser'), ('ETag', '"1-c2e1905c915c668f4395b69025da5746"'), ('Date', 'Wed, 10 Feb 2021 11:50:12 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '80'), ('Cache-Control', 'must-revalidate')]]
 restkit.client.request: DEBUG: Return response: http://localhost:6994/lumi_t/_design/GroupUser
 couchapp.localdoc: INFO: Visit your CouchApp here:
 http://unittestagent:passwd@localhost:6994/lumi_t/_design/GroupUser/index.html
 restkit.client.request: INFO: Start request: HEAD http://localhost:6994/lumi_t
 restkit.client.request: DEBUG: Request headers: [[u'HEAD /lumi_t HTTP/1.1\r\n', 'Host: localhost:6994\r\n', 'User-Agent: couchapp/0.7.1\r\n', 'Accept-Encoding: None\r\n', 'Accept: application/json\r\n', 'Authorization: Basic dW5pdHRlc3RhZ2VudDpwYXNzd2Q=\r\n', '\r\n']]
 restkit.client.request: DEBUG: Start response: 200 OK
-restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Date', 'Wed, 10 Feb 2021 11:06:40 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '230'), ('Cache-Control', 'must-revalidate')]]
+restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Date', 'Wed, 10 Feb 2021 11:50:12 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '230'), ('Cache-Control', 'must-revalidate')]]
 restkit.client.request: DEBUG: Return response: http://localhost:6994/lumi_t
 couchapp.localdoc: DEBUG: push rewrites.json
 couchapp.localdoc: DEBUG: push validate_doc_update.js
@@ -480,7 +431,7 @@
 restkit.client.request: INFO: Start request: GET http://localhost:6994/lumi_t/_design/ACDC
 restkit.client.request: DEBUG: Request headers: [[u'GET /lumi_t/_design/ACDC HTTP/1.1\r\n', 'Host: localhost:6994\r\n', 'User-Agent: couchapp/0.7.1\r\n', 'Accept-Encoding: None\r\n', 'Accept: application/json\r\n', 'Authorization: Basic dW5pdHRlc3RhZ2VudDpwYXNzd2Q=\r\n', '\r\n']]
 restkit.client.request: DEBUG: Start response: 404 Object Not Found
-restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Date', 'Wed, 10 Feb 2021 11:06:40 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '41'), ('Cache-Control', 'must-revalidate')]]
+restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Date', 'Wed, 10 Feb 2021 11:50:12 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '41'), ('Cache-Control', 'must-revalidate')]]
 restkit.client.request: DEBUG: Return response: http://localhost:6994/lumi_t/_design/ACDC
 couchapp.localdoc: DEBUG: attach CouchDBLogo.png 
 couchapp.localdoc: DEBUG: attach DMWMLogo.png 
@@ -500,12 +451,12 @@
 restkit.client.request: INFO: Start request: PUT http://localhost:6994/lumi_t/_design/ACDC
 restkit.client.request: DEBUG: Request headers: [[u'PUT /lumi_t/_design/ACDC HTTP/1.1\r\n', 'Host: localhost:6994\r\n', 'User-Agent: couchapp/0.7.1\r\n', 'Accept-Encoding: None\r\n', 'Accept: application/json\r\n', 'Content-Length: 62364\r\n', 'Content-Type: application/json\r\n', 'Authorization: Basic dW5pdHRlc3RhZ2VudDpwYXNzd2Q=\r\n', '\r\n']]
 restkit.client.request: DEBUG: Start response: 201 Created
-restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Location', 'http://localhost:6994/lumi_t/_design/ACDC'), ('ETag', '"1-36f2444208442143bb721add681464ad"'), ('Date', 'Wed, 10 Feb 2021 11:06:40 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '75'), ('Cache-Control', 'must-revalidate')]]
+restkit.client.request: DEBUG: Response headers: [[('Server', 'CouchDB/1.6.1 (Erlang OTP/R16B03-1)'), ('Location', 'http://localhost:6994/lumi_t/_design/ACDC'), ('ETag', '"1-36f2444208442143bb721add681464ad"'), ('Date', 'Wed, 10 Feb 2021 11:50:12 GMT'), ('Content-Type', 'application/json'), ('Content-Length', '75'), ('Cache-Control', 'must-revalidate')]]
 restkit.client.request: DEBUG: Return response: http://localhost:6994/lumi_t/_design/ACDC
 couchapp.localdoc: INFO: Visit your CouchApp here:
 http://unittestagent:passwd@localhost:6994/lumi_t/_design/ACDC/index.html
 root: INFO: Workflow id 1 created for wf001
-root: INFO: Fileset created: 683d9151-6306-4745-874d-ead52ca2dd03
+root: INFO: Fileset created: 93ad3345-a389-4901-8222-ffc6488143e1
 root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 2}]
 root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 3}]
 root: DEBUG: File.Heritage binds: [{'parent': 1, 'child': 4}]
@@ -525,15 +476,15 @@
 AssertionError: ERROR: Job mask did not load properly
 -------------------- >> begin captured logging << --------------------
 root: DEBUG: Tables for WMCore.WMBS created
-root: INFO: Workflow id 1 created for 5e462018-086c-45b3-b57c-7b95faedf53f
+root: INFO: Workflow id 1 created for e540f966-2f1e-4660-8cb9-dbf24da95a73
 root: INFO: Fileset created: TestFileset
-root: INFO: Fileset created: 2508f07e-9fec-442f-a995-4df042c9f923
+root: INFO: Fileset created: 6cd9bb16-034b-4454-839a-39e66a8362bb
 --------------------- >> end captured logging << ---------------------
 
 ----------------------------------------------------------------------
 XML: /data/cms-dmwm/wmcore_unittest/WMCore/nosetests.xml
 ----------------------------------------------------------------------
-Ran 240 tests in 1581.345s
+Ran 240 tests in 1562.368s
 
-FAILED (SKIP=2, errors=5, failures=4)
+FAILED (SKIP=2, errors=5, failures=3)
 Testing complete, there are now 1 threads
```

**Comment by mapellidario** [comment]
Regarding `Fileset.py`: you are correct. I did not change it. Maybe I was not too clear in the PR description message. 
In `Fileset.py` we use `str()`, so futurize suggested adding `from builtins import str`. Since I would like to change as little code as possible, especially when dealing with strings, I decided not to listen to such suggestions from futurize unless they seem necessary. In this case, I think we are in a good shape not adding anything, but I am not so sure. So, for future-me I added a comment in the PR description mentioning that I willingly did not include such change, betting it will not cause us any problem

File: src/python/WMCore/WMSpec/Persistency.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mapellidario) [reply src/python/WMCore/WMSpec/Persistency.py]: I added a new commit in which we import only urllib from the future backports of the standard library, not affeting pickle/cPickle

File: src/python/WMCore/Services/ReqMgr/ReqMgr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mapellidario) [reply src/python/WMCore/Services/ReqMgr/ReqMgr.py]: Changes added in new commit!

**Comment by amaltaro** [comment]
@mapellidario Dario, if you are no longer working on this PR, can you please squash the src code changes in one commit, and pylint + unit tests in another commit?

It looks good to go from my side (to keep an eye out for those changes in the Persistency module though).

**Comment by mapellidario** [comment]
Reabes onto `master` and squashed. Thanks again!

**Pull request merged**